packages/db/src/search/FileSearch.ts

```diff
@@ -0,0 +1,112 @@
+import { Knex } from 'knex';
+import Logger from '@cumulus/logger';
+import pick from 'lodash/pick';
+import set from 'lodash/set';
+import { ApiFile } from '@cumulus/types';
+import { BaseSearch } from './BaseSearch';
+import { DbQueryParameters, QueryEvent } from '../types/search';
+import { PostgresFileRecord } from '../types/file';
+import { TableNames } from '../tables';
+import { BaseRecord } from '../types/base';
+import { translatePostgresFileToApiFile } from '../translate/file';
+
+const log = new Logger({ sender: '@cumulus/db/ExecutionSearch' });
+
+interface FileRecord extends BaseRecord, PostgresFileRecord{}
+
+/**
+ * Class to build and execute db search query for executions
+ */
+export class FileSearch extends BaseSearch {
+  constructor(event: QueryEvent) {
+    // estimate the table rowcount by default
+    if (event?.queryStringParameters?.estimateTableRowCount !== 'false') {
+      set(event, 'queryStringParameters.estimateTableRowCount', 'true');
+    }
+    super(event, 'file');
+  }
+
+  /**
+   * check if joined granules table search is needed
+   *
+   * @returns whether granules table search is needed
+   */
+  protected searchGranules(): boolean {
+    const { not, term, terms } = this.dbQueryParameters;
+    return (!!(not?.granuleCumulusId || term?.granuleCumulusId || terms?.granuleCumulusId));
+  }
+
+  /**
+   * Build basic query
+   *
+   * @param knex - DB client
+   * @returns queries for getting count and search result
+   */
+  protected buildBasicQuery(knex: Knex)
+    : {
+      countQuery: Knex.QueryBuilder,
+      searchQuery: Knex.QueryBuilder,
+    } {
+    const {
+      granules: granulesTable,
+    } = TableNames;
+
+    const searchQuery = knex(`${this.tableName}`)
+      .select(`${this.tableName}.*`)
+      .select({
+        collectionName: `${granulesTable}.name`,
+      });
+
+    const countQuery = knex(this.tableName)
+      .count('*');
+
+    if (this.searchGranules()) {
+      countQuery.innerJoin(granulesTable, `${this.tableName}.granule_cumulus_id`, `${granulesTable}.cumulus_id`);
+      searchQuery.innerJoin(granulesTable, `${this.tableName}.granule_cumulus_id`, `${granulesTable}.cumulus_id`);
+    } else {
+      searchQuery.leftJoin(granulesTable, `${this.tableName}.granule_cumulus_id`, `${granulesTable}.cumulus_id`);
+    }
+    return { countQuery, searchQuery };
+  }
+
+  /**
+   * Build queries for infix and prefix
+   *
+   * @param params
+   * @param params.countQuery - query builder for getting count
+   * @param params.searchQuery - query builder for search
+   * @param [params.dbQueryParameters] - db query parameters
+   */
+  protected buildInfixPrefixQuery(params: {
```

Comment:
This appears to be a boilerplate copy, *other* than the field name.    Not that it has to be done here, but bringing it up since the pattern is propagating here - does it really make sense to duplicate this much boilerplate to differ between `.name` and, say `.granule_id`?

Reply:
this was entirely a mistake, not meant to be committed, from my early "do I need a FIleSearch"
removed